core/modules/image/image.field.inc

```diff
@@ -200,6 +204,23 @@ function image_field_instance_settings_form($field, $instance) {
   return $form;
 }
 
+/**
+ * #element_validate handler for image_field_instance_settings_form().
+ */
+function image_field_settings_form_validate($element, &$form_state, $complete_form) {
+  $max_dimensions_x = $form_state['input']['instance']['settings']['max_dimensions']['x'];
+  $max_dimensions_y = $form_state['input']['instance']['settings']['max_dimensions']['y'];
+  $min_dimensions_x = $form_state['input']['instance']['settings']['min_dimensions']['x'];
+  $min_dimensions_y = $form_state['input']['instance']['settings']['min_dimensions']['y'];
```

Comment:
We should be using `$form_state['values']` rather than `$form_state['input']`. The "values" are checked by Drupal to make sure corresponding elements exist, where "input" can be anything sent by the user through a `POST` request, even things that don't have form elements, so using "values" is safer.